### PR TITLE
Remove workarounds used during D2 transition

### PR DESCRIPTION
**Pull request by don-clugston-sociomantic**
This PR cleans up the main workarounds used during D2 transition.
These are the code changes required before `import ocean.transition` can be removed (the remaining required changes only involve changes to import statements).
None of the changes in this PR actually affect any generated code.

### CODE REVIEW

File: src/ocean/util/Convert.d [hunk 1]
```diff
@@ -916,7 +916,8 @@ D toStringFromString(D,S)(S value)
     static if (is(S : D))
         return value;
 
-    else static if (!isMutable!(DElem) || !isMutable!(SElem))
```
Comment (Geod24) [inline src/ocean/util/Convert.d:919]: `isMutable` still looks better TBH

File: src/ocean/math/Range.d [hunk 2]
```diff
@@ -215,6 +215,8 @@ public struct Range ( T )
     {
         public istring toString()
         {
+            import ocean.core.TypeConvert: assumeUnique;
```
Comment (jens-mueller-sociomantic) [inline src/ocean/math/Range.d:218]: This seems unrelated to the commit. Looks like you removed an unnecessary import here. But I don't see by which import it is actually covered.

**Review by jens-mueller-sociomantic — COMMENTED** [review]
Looks good to me. The only thing that confuse me is the removal of this one import statement.

File: src/ocean/util/Convert.d (hunk 1 above)
Comment (jens-mueller-sociomantic) [reply src/ocean/util/Convert.d:919]: True, what were the reasons to drop it?

Comment (don-clugston-sociomantic) [reply src/ocean/util/Convert.d:919]: You reckon? Personally I don't like the double negative (if it is not not const and not not immutable ...) But the point is, the use of `isMutable`  is a blocker to removing ocean.transition, it doesn't exist elsewhere in ocean.

Comment (Geod24) [reply src/ocean/util/Convert.d:919]: Maybe `isMutable` should be made a type traits then ?
Or use a different condition, such as `is(Unqual!T == T)` to show that only unqualified types are accepted ?

File: src/ocean/math/Range.d (hunk 2 above)
Comment (don-clugston-sociomantic) [reply src/ocean/math/Range.d:218]: This was the only remaining use of `ocean.transition` in this module, which is the reason for making these changes. It will be removed from everywhere else in ocean in the next PR.

File: src/ocean/util/Convert.d (hunk 1 above)
Comment (don-clugston-sociomantic) [reply src/ocean/util/Convert.d:919]: I do not think anything like that is justified for a single expression which is only used in one place in the internals of a function. I am trying to make the absolutely minimal changes required to finish the ocean D2 transition. I can't justify anything more complicated than that.

**Review by Geod24 — APPROVED** [review]
LGTM
